pandas/core/arrays/categorical.py

```diff
@@ -2324,14 +2337,20 @@ def replace(self, to_replace, value, inplace: bool = False):
                 continue
             if replace_value in cat.categories:
                 if isna(new_value):
-                    cat.remove_categories(replace_value, inplace=True)
+                    with catch_warnings():
+                        simplefilter("ignore")
+                        cat.remove_categories(replace_value, inplace=True)
```

Comment:
i think eventually we'll need to push this up and deprecate inplace here in Categorical.replace, but that can be done in a separate step